docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -352,13 +352,145 @@ quarkus.http.auth.policy.role-policy-cert.roles-allowed=user,admin     <2>
 
 Given the preceding configuration, the request is authorized if the client certificate's CN attribute is equal to `alice` or `bob` and forbidden if it is equal to `jdoe`.
 
+[[use-cert-attrs-to-augment-identity]]
 ==== Using certificate attributes to augment SecurityIdentity
 
 You can always register `SecurityIdentityAugmentor` if the automatic <<map-certificate-attributes-to-roles>> option does not suit.
 Custom `SecurityIdentityAugmentor` can check the values of different client certificate attributes and augment the `SecurityIdentity` accordingly.
 
 For more information about customizing `SecurityIdentity`, see the xref:security-customization.adoc#security-identity-customization[Security identity customization] section in the Quarkus "Security tips and tricks" guide.
 
+[[mtls-programmatic-set-up]]
+==== Set up the mutual TLS client authentication programmatically
+
+The `io.quarkus.vertx.http.security.HttpSecurity` CDI event allows to configure the mutual TLS client authentication mechanism programmatically.
+Let's consider the example explained in the <<map-certificate-attributes-to-roles>> section:
+
+[source,properties]
+----
+quarkus.tls.tls-config-1.key-store.p12.path=server-keystore.p12
+quarkus.tls.tls-config-1.key-store.p12.password=the_key_store_secret
+quarkus.tls.tls-config-1.trust-store.p12.path=server-truststore.p12
+quarkus.tls.tls-config-1.trust-store.p12.password=the_trust_store_secret
+quarkus.http.ssl.client-auth=required
+quarkus.http.insecure-requests=disabled
+
+quarkus.http.auth.certificate-role-properties=cert-role-mappings.properties
+
+quarkus.http.auth.permission.certauthenticated.paths=/*
```

Comment:
Can we drop the role mapping here ? As this section introduces enabling the MTLS authentication